File: Modules/Common/Techs/SIE.Web.Tech/Routings/Commands/DeleteRouting.js
SIE.defineCommand('SIE.Web.Tech.Routings.Commands.DeleteRouting', {
    extend: 'SIE.cmd.Command',
    meta: { text: "", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    execute: function (view, source) {
        var routingNode = source;
        var treePanel = routingNode.getOwnerTree();
        var routingId = routingNode.get("Id");
        var command = Ext.getClassName(this);
        //客制化界面命令注册签名（注册在context，没有后台请求）
        SIE.Signature.createCmdContext({ command: command, commandName: "删除", Type: view.model });
        SIE.Msg.confirm("确定要删除选择的工艺路线吗?".L10N(), function () {
            view.execute({
                command: command,
                data: routingId,
                success: function (res) {
                    routingNode.parentNode.removeChild(routingNode);
                    if (view.routingId === routingId) {
                        view.layout.designControl.resetMainBlock();
                    }
                }
            });
        });
    }
});

File: Modules/Common/Techs/SIE.Web.Tech/Routings/Commands/DeleteRouting.cs
using SIE.Domain;
using SIE.Security;
using SIE.Tech.Routings;
using SIE.Web.Command;
using System;

namespace SIE.Web.Tech.Routings.Commands
{
    /// <summary>
    /// 删除工艺路线命令
    /// </summary>
    [JsCommand("SIE.Web.Tech.Routings.Commands.DeleteRouting")]
    [AllowAnonymous]
    public class DeleteRouting : ViewCommand
    {
        /// <summary>
        /// 删除工艺路线命令执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var routingId = Convert.ToDouble(args.Data);
            var routing = RF.GetById<Routing>(routingId);
            routing.PersistenceStatus = PersistenceStatus.Deleted;
            RF.Save(routing);
            return true;
        }
    }
}